controllers/chat/moveToSign.js

```diff
@@ -0,0 +1,107 @@
+const {User, ChatAnonUserInfo} = require('../../databases/models');
+const {StreamChat} = require('stream-chat');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const {responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const moveToSign = async (req, res) => {
+  const {targetUser, oldChannelId} = req.body;
+
+  let members = [targetUser];
+  if (!members.includes(req.userId)) members.push(req.userId);
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    const {userModel, targetUserModel} = await Promise.allSettled([
+      UsersFunction.findUserById(User, req?.userId),
+      UsersFunction.findUserById(User, targetUser)
+    ]);
+
+    /**
+     * @type {import('stream-chat').OwnUserResponse}
+     */
+    const user = {
+      name: userModel?.username,
+      id: req?.userId,
+      image: userModel?.profile_pic_path,
+      username: userModel?.username
+    };
+
+    await client.connectUser(user, req.token);
+
+    const newChannel = client.channel('messaging', {members});
+    const createdChannel = await newChannel.create();
+
+    const targets = members.filter((member) => member !== req.userId);
+    await targets.map(async (target) => {
+      const userModel = await UsersFunction.findUserById(User, target);
+
+      if (userModel.is_anonymous) {
+        const checkChatAnonUserInfo = await ChatAnonUserInfo.findOne({
+          where: {
+            channel_id: oldChannelId,
+            my_anon_user_id: req.userId,
+            target_user_id: target
+          }
+        });
+
+        if (checkChatAnonUserInfo !== null) {
+          await ChatAnonUserInfo.create({
+            channel_id: newChannel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target,
+            anon_user_info_color_code: checkChatAnonUserInfo?.anon_user_info_color_code,
+            anon_user_info_color_name: checkChatAnonUserInfo?.anon_user_info_color_name,
+            anon_user_info_emoji_code: checkChatAnonUserInfo?.anon_user_info_emoji_code,
+            anon_user_info_emoji_name: checkChatAnonUserInfo?.anon_user_info_emoji_name
+          });
+        }
+      }
+    });
+
+    try {
+      if (!newChannel?.data?.name) {
+        await newChannel.updatePartial({
+          set: {
+            channel_type: CHANNEL_TYPE.CHAT,
```

Comment:
Should check if any of the members is anonymous, the channel_type should be CHANNEL_TYPE.ANONYMOUS